railties/test/application/rake/dbs_test.rb

```diff
@@ -687,35 +687,54 @@ def db_fixtures_load(expected_database)
         end
       end
 
-      test "db:prepare setup the database" do
+      test "db:prepare loads schema, runs pending migrations, and updates schema" do
         Dir.chdir(app_path) do
           rails "generate", "model", "book", "title:string"
           output = rails("db:prepare")
           assert_match(/CreateBooks: migrated/, output)
+          assert_match(/create_table "books"/, File.read("db/schema.rb"))
 
           output = rails("db:drop")
           assert_match(/Dropped database/, output)
 
           rails "generate", "model", "recipe", "title:string"
           output = rails("db:prepare")
-          assert_match(/CreateBooks: migrated/, output)
+          assert_no_match(/CreateBooks: migrated/, output) # loaded from schema
           assert_match(/CreateRecipes: migrated/, output)
+
+          schema = File.read("db/schema.rb")
+          assert_match(/create_table "books"/, schema)
+          assert_match(/create_table "recipes"/, schema)
+
+          tables = rails("runner", "p ActiveRecord::Base.connection.tables.sort").strip
+          assert_equal('["ar_internal_metadata", "books", "recipes", "schema_migrations"]', tables)
         end
       end
 
-      test "db:prepare does not touch schema when dumping is disabled" do
+      test "db:prepare loads schema when database exists but is empty" do
+        rails "generate", "model", "book", "title:string"
+        rails("db:prepare", "db:drop", "db:create")
+
+        output = rails("db:prepare")
+        assert_no_match(/CreateBooks: migrated/, output)
+
+        tables = rails("runner", "p ActiveRecord::Base.connection.tables.sort").strip
+        assert_equal('["ar_internal_metadata", "books", "schema_migrations"]', tables)
+      end
+
+      test "db:prepare does not dump schema when dumping is disabled" do
         Dir.chdir(app_path) do
           rails "generate", "model", "book", "title:string"
           rails "db:create", "db:migrate"
 
-          app_file "db/schema.rb", "Not touched"
+          app_file "db/schema.rb", "# Not touched"
```

Comment:
I think the syntax error here was intentional. The test should raise if the file is ever eval'd. See https://github.com/rails/rails/pull/36744, "this is causing issues in environments with limited file permissions that are preparing database."

Reply:
@morgoth correct me if I'm wrong!

Reply:
I think `touch` is a bit ambiguous, but my reading of that PR is to avoid _writing_ ("dumping") to the `schema.rb` file, not to avoid _reading_ from it. 

Even in that PR, the behavior would still load the schema file, but only if the database didn't exist. Which was not a scenario covered in the test cases.

Reply:
My intention was only to test if the file content is preserved.